### ISSUE TITLE
discrete_approx_mvn (sample-ghmm) is kind of slow

### ISSUE THREAD

**Issue by rmcgibbo** [msmbuilder/msmbuilder#54]
- BFGS optimization over the lagrange multipliers on the constraints. Not clear that it can be hot-started any better.
- Stochastic gradient descent?
- Profile the performance of the objective function / gradient calculation in detail
- Possible things that could help: 
  - Call sklearn.utils.ext_math.fast_dot, which -- depending on the version of numpy -- link against a faster blas.
  - Write the inner loop for computing the obj / grad in C.
